main/migrations/sqls/20210208213802-AddDomainEventTable-up.sql

```diff
@@ -0,0 +1,24 @@
+/* Replace with your SQL commands */
+CREATE TABLE domain_event
+(
+    id uuid NOT NULL,
+    payload jsonb NOT NULL,
+    status varchar NOT NULL,
+    created_at timestamptz NOT NULL,
+    updated_at timestamptz NOT NULL,
+    CONSTRAINT domain_event_pkey PRIMARY KEY (id, status, created_at)
+) PARTITION BY LIST (status);

```

Comment:
Creating partitions by status values, we expect the domain events to have `raised`, `sent`, `received`, `handled` values.
Of these, `raised` and `received` will be short lived and they will transition to `sent` and `handled` when processed.
Thus, a large portion of data will be in `sent` and `handled` state. The partitions for those are then sub-partitioned by the created_at so that we can remove the old partitions easily.